Pegasus/include/Math.hpp

```diff
@@ -1,26 +1,655 @@
 /*
-* Copyright (c) Icosagon 2003. All Rights Reserved.
-*
-* This software is distributed under licence. Use of this software
-* implies agreement with all terms and conditions of the accompanying
-* software licence.
+* Copyright (C) 2017 by Godlike
+* This code is licensed under the MIT license (MIT)
+* (http://opensource.org/licenses/MIT)
 */
 #ifndef PEGASUS_MATH_HPP
 #define PEGASUS_MATH_HPP
 
+#define GLM_ENABLE_EXPERIMENTAL
+#include <glm/gtx/norm.hpp>
 #include <glm/glm.hpp>
 
+#include <iterator>
+#include <vector>
+#include <array>
+#include <list>
+#include <set>
+#include <unordered_map>
+#include <unordered_set>
+#include <algorithm>
+#include <utility>
+
 namespace pegasus
 {
 namespace math
 {
+/**
+ * @brief HyperPlane calculation algorithm.
+ */
+class HyperPlane
+{
+public:
+    HyperPlane() = default;
+
+    /**
+     * @brief Constructs a plane in Hessian Normal Form.
+     * @param normal normal plane vector
+     * @param point point on the plane
+     * @param below point below the plane
+     */
+    HyperPlane(glm::dvec3 const& normal,
+        glm::dvec3 const& point,
+        glm::dvec3 const* below = nullptr
+    );
+
+    /**
+     * @brief Constructs a plane in Hessian Normal Form.
+     * @param a point on a plane
+     * @param b point on a plane
+     * @param c point on a plane
+     * @param below point below plane
+     */
+    HyperPlane(glm::dvec3 const& a,
+        glm::dvec3 const& b,
+        glm::dvec3 const& c,
+        glm::dvec3 const* below = nullptr
+    );
+
+    /**
+     * @brief Constructs a plane in Hessian Normal Form.
+     * @param vertices points on a plane
+     * @param below point below plane
+     */
+    HyperPlane(
+        glm::dmat3 const& vertices,
+        glm::dvec3 const* below
+    );
+
+    /**
+     * @brief Plane point getter.
```

Comment:
Don't write what it **is**, write what it **does**. For getters and setters you can have oneline comments, for example `Returns a point on the plane`